cmd/node/list_test.go

```diff
@@ -0,0 +1,114 @@
+package node
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/cilium/cilium/api/v1/relay"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cilium/cilium/api/v1/observer"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_jsonOutput(t *testing.T) {
+	nodes := []*observer.Node{
+		{
+			Name:    "foo",
+			Version: "1",
+		},
+		{
+			Name:    "bar",
+			Version: "2",
+		},
+		{
+			Name:    "baz",
+			Version: "3",
+		},
+	}
+
+	buf := bytes.Buffer{}
+	cmd := cobra.Command{}
+	cmd.SetOut(&buf)
+
+	require.NoError(t, jsonOutput(&cmd, nodes))
+	assert.Equal(t, `[
+  {
+    "name": "foo",
+    "version": "1"
+  },
+  {
+    "name": "bar",
+    "version": "2"
+  },
+  {
+    "name": "baz",
+    "version": "3"
+  }
+]
+`, buf.String())
+}
+
+func Test_tableOutput(t *testing.T) {
+	defer func() {
+		listOpts.output = ""
+	}()
+
+	var testCases = []struct {
+		name           string
+		tableType      string
+		expectedOutput string
+	}{
+		{
+			name:      "happy path with regular table output",
+			tableType: "table",
+			expectedOutput: `NAME   STATUS        AGE      FLOWS/S   CURRENT/MAX-FLOWS
+foo    Connected     16m40s   0.00      500/1000 ( 50.00%)
+bar    Unavailable   N/A      N/A       N/A
+`,
+		},
+		{
+			name:      "happy path with wide table output",
+			tableType: "wide",
+			expectedOutput: `NAME   STATUS        AGE      FLOWS/S   CURRENT/MAX-FLOWS    VERSION   ADDRESS   TLS
+foo    Connected     16m40s   0.00      500/1000 ( 50.00%)   1         1.2.3.4   Enabled
+bar    Unavailable   N/A      N/A       N/A                  2         5.6.7.8   N/A
+`,
+		},
+	}
+
+	for _, tc := range testCases {
+		listOpts.output = tc.tableType
+
+		nodes := []*observer.Node{
```

Comment:
I think that `nodes` should be part of the `testCases` struct as tests are tightly coupled with it.